azkaban-common/src/main/java/azkaban/utils/FileIOUtils.java

```diff
@@ -69,6 +71,51 @@ public static boolean isDirWritable(final File dir) {
     return true;
   }
 
+
+  /**
+   * Returns last modified time of file.
+   */
+  public static long getLastModifiedTime(final File file) throws IOException {
+    Preconditions.checkArgument(file.exists(), file + " doesn't exist");
+    return file.lastModified();
+  }
+
+  /**
+   * Dumps a number into a new file.
+   *
+   * @param filePath the target file
+   * @param num the number to dump
+   * @throws IOException if file already exists
+   */
+  public static void dumpNumberToFile(final String filePath, final long num) throws IOException {
```

Comment:
Have you considered accepting a https://docs.oracle.com/javase/8/docs/api/java/nio/file/Path.html instead? You can benefit from stronger type checking.

Reply:
And it will be easier to unit test.

Reply:
updated, thanks